main.js

```diff
@@ -0,0 +1,90 @@
+// var posts = [];
+var counter = 0;
+
+$("#post-form").submit(function(event) {
+  counter += 1;
+
+  var userPost = $("#message").val()
+  var userName = $("#name").val()
+  // Decided not to go with an object implementation.
+  // var post = {
+  //   post: userPost,
+  //   name: userName
+  // }
+  // posts.push(post);
+  var postDiv = createPost(userPost, userName, counter);
+
+  $('.posts').append(postDiv);
+
+  $('#message').val('');
+  $('#name').val('');
+
+  event.preventDefault();
+});
+
+// Create a function to create and append a template with the variables
+// from the posts form. Attach click events to each dynamic created elements.
+var createPost = function(post, name, counter) {
+  userPost = `<p data-post-id="${counter}" ><span class="post-delete label label-danger"><a href="#">remove</a></span> <span class="comments label label-info"><a href="#">comments</a></span> -${post}</p>`;
+  userName = `<p data-name-id="${counter}" ><small>Posted By:</small> <b class="users">${name}</b></p>`;
+  var template = `
+  <div class="post" id="post-${counter}" >${userPost}
+    <div class="comment-section"></div>
+    <form class="comment-form" id='${counter}'>
+      <div class="form-row align-items-center">
+        <div class="col-sm-4 my-1">
+          <input type="text" class="form-control" id="comment-text${counter}" placeholder="Comment Text">
+        </div>
+        <div class="col-sm-4 my-1">
+          <input type="text" class="form-control" id="user-text${counter}" placeholder="User Name">
+        </div>
+        <div class="col-auto my-1">
+          <button type="button" class="btn btn-primary comment-post">Submit</button>
+        </div>
+      </div>
+    </form>
+    ${userName}<hr>
+  </div>`
+
+  var handleDeletePost = function() {
```

Comment:
Why did you put all of these other functions inside of `createPost`? If it's to register handlers on elements that don't yet exist in the DOM, it would be cleaner to use jQuery's `on` method.